### PR TITLE
Run2-hcx185 Treat the subdetid method correctly

### PR DESCRIPTION
**Pull request by bsunanda**
Also avoid warning statements during compilation

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@kpedro88 I wonder how this PR can fail the tests. They are always inability to open files.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @civanch @lpernie @rekovic Could you please check & approve this bug fix PR?

**Comment by bsunanda** [comment]
@ianna @civanch @rekovic Could you please sign this PR?

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by rekovic** [comment]
+1

File: DataFormats/HcalDetId/interface/HcalDetId.h [hunk 1]
```diff
@@ -13,22 +13,22 @@
...
-  static const int kHcalZsideMask2     = 0x80000;
-  static const int kHcalDepthOffset1   = 14;
-  static const int kHcalDepthOffset2   = 20;
-  static const int kHcalDepthMask1     = 0x1F;
-  static const int kHcalDepthMask2     = 0xF;
-  static const int kHcalDepthSet1      = 0x1C000;
-  static const int kHcalDepthSet2      = 0xF00000;
-  static const int kHcalIdFormat2      = 0x1000000;
-  static const int kHcalIdMask         = 0xFE000000;
+  static const uint32_t kHcalPhiMask1       = 0x7F;
```
Comment (fabiocos) [inline DataFormats/HcalDetId/interface/HcalDetId.h]: @bsunanda @Dr15Jones I assume these declarations could become constexpr uint32_t, although not necessary

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @civanch @lpernie @rekovic Could you sign this PR yet again?

**Comment by lpernie** [comment]
+1

**Comment by thomreis** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
close this PR in favour of #24917

**Pull request closed without merging**